### DISCUSSION TITLE
Debug output button presses

### DISCUSSION THREAD

**Discussion by ledah90** [darthcloud/BlueRetro#194] [Q&A] [answered]
Is there a way to see what button I am pressing on a custom bluetooth controller over serial or the live output feed on the website? I have a n64 controller that I made and its mapping is all over the place. I can't seem to figure out exactly what button blueretro is seeing when I press it. If I connect via serial, I'm only getting bluetooth connection output and on/off status. Connecting my controller in windows, it functions normally and it shows as a 16 button, 2 axis controller.

**Comment by darthcloud** [comment, marked as answer]
Flash BlueRetro_pad_test.bin from v1.3 and set your terminal as in : https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

"Implicit CR in every LF" is important.

**Reply by darthcloud** [reply]
BTW this ignore any custom blueretro mapping, it show the "source"

**Reply by ledah90** [reply]
Is the custom mapping like "BD, DD, BL, DU"? This serial out is giving me buttons like "BD, DD, BL, DU", but when I go to the config page, there are no buttons matching any of those labels, except a couple.

**Reply by darthcloud** [reply]
It's the shortened form RB \<direction> = B\<direction first letter>  and D-pad \<direction> = D\<direction first letter>

**Reply by ledah90** [reply]
I got all of my buttons mapped, but I'm having issues getting the joystick programmed. It acts like I'm always moving in a specific direction and Up doesn't work at all. Any pointers on figuring out proper joystick calibration and settings? When using the controller in windows, it always returns to 0,0 when not moving the joystick.

**Reply by darthcloud** [reply]
joystick value are unsigned, if you report 16bit value: so zero is left, 0xFFFF is right, 0x8000 is neutral.
